pkgs/development/python-modules/osqp/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, pythonOlder
+, buildPythonPackage
+, fetchPypi
+, cmake
+, future
+, numpy
+  # check inputs
+, scipy
+, pytestCheckHook
+, mkl
+}:
+
+buildPythonPackage rec {
+  pname = "osqp";
+  version = "0.6.1";
+
+  disabled = pythonOlder "3.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "130frig5bznfacqp9jwbshmbqd2xw3ixdspsbkrwsvkdaab7kca7";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  dontUseCmakeConfigure = true;
+
+  propagatedBuildInputs = [
+    numpy
+    future
+  ];
+
+  doCheck = true;
+  checkInputs = [ scipy pytestCheckHook mkl ];
+  dontUseSetuptoolsCheck = true;
```

Comment:
there's no check phase then... please add tests

Reply:
``pytestCheckHook`` runs pytest automatically. Setuptools Check doesn't work with several of these packages.

Reply:
This line doesn't specifically need it, but it still uses pytestCheckHook. Fixed in commit.